examples/pwelch.cu

```diff
@@ -0,0 +1,105 @@
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "matx.h"
+#include <cassert>
+#include <cstdio>
+#include <cuda/std/ccomplex>
+
+using namespace matx;
+
+/**
+ * PWelch
+ *
+ * This example shows how to estimate the power spectral density of a 1D tensor
+ * using Welch's method [1].
+ *
+ * [1] P. Welch, "The use of fast Fourier transform for the estimation of power spectra: A method based on time averaging over short, modified periodograms," in IEEE Transactions on Audio and Electroacoustics, vol. 15, no. 2, pp. 70-73, June 1967, doi: 10.1109/TAU.1967.1161901.
+ *
+ */
+
+int main([[maybe_unused]] int argc, [[maybe_unused]] char **argv)
+{
+  MATX_ENTER_HANDLER();
+  using complex = cuda::std::complex<float>;
+
+  float exec_time_ms;
+  const int num_iterations = 100;
+  index_t signal_size = 256;
+  int nperseg = 32;
+  int nfft = nperseg;
+  int noverlap = 8;
+  float ftone = 3.0;
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+  cudaEvent_t start, stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+
+  // Create buffers
+  auto x  = make_tensor<complex>({signal_size});
+  auto Pxx  = make_tensor<typename complex::value_type>({nfft});
+
+  for (index_t k = 0; k < signal_size; k++) {
+    // Create the time domain signal as a complex exponential
+    float phase = static_cast<float>(2.0*M_PI*ftone*static_cast<float>(k)/nfft);
+    x(k) = {static_cast<float>(cos(phase)), static_cast<float>(sin(phase))};
+  }
+
+  // Prefetch the data we just created
+  x.PrefetchDevice(0);
```

Comment:
In the future we can leave these out of examples since they don't really add anything to it